src/Rules/NodeVisitorRuleInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace TwigCsFixer\Rules;
+
+use TwigCsFixer\Report\Report;
+
+interface NodeVisitorRuleInterface
+{
+    /**
+     * Messages will be added to the given `$report` object.
+     */
+    public function enterFile(Report $report, string $filePath): void;
```

Comment:
Funny, this is what I initially did. But I didn't like the fact that `$this->report` could be nullable. 

Also, I feel like when you set it, you should also unset on leaving the file.

That reminds me, I should make sure `$this->violations` is cleared as well. 

Up to you, what you think is better. I will change it.

Reply:
In the same way I introduce an AbstractRule to help writing rules, I feel like an AbstractNodeVisitorRule could be helpful and may help to decide what the best strategy.

`AbstractRule::addMessage` might be in a trait (or copy pasted) and should be in `AbstractNodeVisitorRule` too

I'd like to simplify the code to
```
public function enterNode(Node $node, Environment $env): Node
    {
        if (!$node instanceof FilterExpression) {
            return $node;
        }

        if (!\in_array($node->getNode('filter')->getAttribute('value'), $this->forbidden, true)) {
            return $node;
        }

        $this->addError(...);

        return $node;
    }
```
like in classic rules.

By the way, `addMessage` is handling `ignoredViolations` and we should have that too, so the enter file method will need the signature
```
public function enterFile(Report $report, string $filePath, array $ignoredViolations = []): void;
```

Since `private ?Report $report = null;` works fine on AbstractRule, I think it can be the same on `AbstractNodeVisitorRule`.